File: engine/app/models/alerts.py
from typing import Optional

from pydantic import BaseModel, Field


class AlertOut(BaseModel):
    id: str
    store_id: Optional[str] = None
    module: str
    severity: str
    title: str
    message: str
    is_read: bool = False
    created_at: str


class CreateAlertRequest(BaseModel):
    store_id: Optional[str] = None
    module: str = Field(..., min_length=1)
    severity: str = Field(..., pattern=r"^(info|warning|error|critical)$")
    title: str = Field(..., min_length=1, max_length=200)
    message: str = Field(..., min_length=1)


class MarkReadRequest(BaseModel):
    alert_ids: list[str]

File: desktop/src/types/alerts.ts
export interface Alert {
  id: string;
  store_id: string | null;
  module: string;
  severity: 'info' | 'warning' | 'error' | 'critical';
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export interface AlertsPage {
  items: Alert[];
  total: number;
  page: number;
  page_size: number;
}
